### PR TITLE
Make storybook configurable in a redwood app 

### PR DESCRIPTION
**Pull request by jtoar**
This PR makes storybook's `main.js` file configurable from a Redwood App by copying the strategy we currently use with webpack. The webpack-merge docs say that the merge function works with more than just webpack configs, so I reused it here. [Here's the quote from the docs](https://github.com/survivejs/webpack-merge#webpack-merge---merge-designed-for-webpack):

> This behavior is particularly useful in configuring webpack although it has uses beyond it. Whenever you need to merge configuration objects, webpack-merge can come in handy.

This PR doesn't do anything for `storybook/preview.js` yet&mdash;I know that's what https://github.com/redwoodjs/redwood/issues/989 was mainly about. So a question while we're on that topic:

- [x] should we just have one `storybook.config.js` file in a redwood app, with a bunch of exports (one for main, one for preview, etc) or a storybook config dir, like we have in core?&
  - **answer**: two files, one suffixed with `.config.js`, the other with `.preview.js`

Finally, here's an unimpressive screenshot just to show that things are working:

image

The file in the redwood app looks like:

```js
// web/config/storybook.config.js

module.exports = {
  addons: ['@storybook/addon-a11y'],
}
```

### CODE REVIEW

**Comment by peterp** [comment]
@jtoar I think it makes sense to have two separate configuration files, since the one has a Node.js context and the other has a browser context. I don't think they have to be in a directory, so maybe just `config/storybook.preview.js` and `config/storybook.config.js`

**Comment by peterp** [comment]
@jtoar I think it makes sense to have two separate configuration files, since the one has a Node.js context and the other has a browser context. I don't think they have to be in a directory, so maybe just `config/storybook.preview.js` and `config/storybook.config.js`

**Review by peterp — CHANGES_REQUESTED** [review]
Here's my comment from a few days ago, I wasn't sure if you saw it.

I do not think that we need to do `storybook.preview.js` right now. If you agree then we can merge this and add `storybook.preview.js` seperately.

> @jtoar I think it makes sense to have two separate configuration files, since the one has a Node.js context and the other has a browser context. I don't think they have to be in a directory, so maybe just config/storybook.preview.js and config/storybook.config.js

File: packages/core/config/storybook/main.js [hunk 1]
```diff
@@ -17,8 +15,13 @@ module.exports = {
...
     // We replace imports to "@redwoodjs/router" with our own implementation in "@redwoodjs/testing"
-    sbConfig.resolve.alias['@redwoodjs/router$'] = path.join(getPaths().base, 'node_modules/@redwoodjs/testing/dist/MockRouter.js')
-    sbConfig.resolve.alias['~__REDWOOD__USER_ROUTES_FOR_MOCK'] = getPaths().web.routes
+    sbConfig.resolve.alias['@redwoodjs/router$'] = path.join(
+      getPaths().base,
+      'node_modules/@redwoodjs/testing/dist/MockRouter.js'
+    )
+    sbConfig.resolve.alias[
+      '~__REDWOOD__USER_ROUTES_FOR_MOCK'
+    ] = getPaths().web.routes
```
Comment (peterp) [inline packages/core/config/storybook/main.js:24]: Prettier did this?

File: packages/internal/src/paths.ts [hunk 2]
```diff
@@ -31,6 +31,7 @@ export interface BrowserTargetPaths {
   config: string
   webpack: string
   postcss: string
+  storybook: string
```
Comment (peterp) [inline packages/internal/src/paths.ts:34]:
```suggestion
  storybookConfig: string
```

Think ahead to when we need to add `storybookPreview`

File: packages/core/config/storybook/main.js (hunk 1 above)
Comment (jtoar) [reply packages/core/config/storybook/main.js:24]: I think so, I definitely didn't do it manually. Happy to revert it if you want; not getting a linting error in my editor right now the way it is. Could I have configured the settings differently somehow?

**Comment by jtoar** [comment]
Dang, screwed up the rebasing again; hold!

File: packages/core/config/storybook/main.js (hunk 1 above)
Comment (jtoar) [reply packages/core/config/storybook/main.js]: Tried putting it back to how it was and I wasn't getting a linting error, so reverted it!

**Comment by jtoar** [comment]
> I do not think that we need to do storybook.preview.js right now. If you agree then we can merge this and add storybook.preview.js seperately.


@peterp totally agree, let's get this one in for the next release if you think it's looking good!

**Review by peterp — APPROVED** [review]

**Comment by peterp** [comment]
Ship it!

**Comment by thedavidprice** [comment]
@jtoar thoughts about where/how to document this?

**Comment by jtoar** [comment]
@thedavidprice time for a Storybook doc I think; I'lll add it as a section there, then link to some of the other docs we have like Mocking GraphQL Requests.

**Comment by jangxyz** [comment]
@peterp 's [change](https://github.com/redwoodjs/redwood/pull/1828/commits/282a34708bf44328f51edb4ad4872870a7d6b423#r584123492) breaks loading user storybook config in the current release (`0.27.1`). Currently it's still reading from `redwoodPaths.web.storybook`. ([core/config/storybook/main.js#L64](https://github.com/redwoodjs/redwood/blob/v0.27.1/packages/core/config/storybook/main.js#L64))

Maybe you should apply the change to `main.js` as well?

**Comment by jtoar** [comment]
That was my bad, and the fix is here: https://github.com/redwoodjs/redwood/pull/2002. Looks ok to you @jangxyz?

**Comment by jangxyz** [comment]
Looks great! 😎

**Pull request merged**